packages/account-abstraction/src/BaseAccountAPI.ts

```diff
@@ -218,41 +224,51 @@ export abstract class BaseAccountAPI {
   }
 
   async encodeUserOpCallDataAndGasLimit(
-    detailsForUserOp: TransactionDetailsForUserOp
+    detailsForUserOp: TransactionDetailsForBatchUserOp
   ): Promise<{ callData: string; callGasLimit: BigNumber }> {
     /* eslint-disable  @typescript-eslint/no-explicit-any */
     function parseNumber(a: any): BigNumber | null {
       if (a == null || a === '') return null
       return BigNumber.from(a.toString())
     }
 
-    if (detailsForUserOp && detailsForUserOp.target === '' && detailsForUserOp.data === '') {
-      return {
-        callData: '0x',
-        callGasLimit: BigNumber.from('21000')
+    let callData
+    if ( detailsForUserOp.target.length == 1 )
+    {
+      if (detailsForUserOp && detailsForUserOp.target[0] === '' && detailsForUserOp.data[0] === '') {
+        return {
+          callData: '0x',
+          callGasLimit: BigNumber.from('21000')
+        }
       }
+      const value = parseNumber(detailsForUserOp.value[0]) ?? BigNumber.from(0)
+      callData = await this.encodeExecuteCall(
```

Comment:
if wallet is not deployed, if there is good way we would be using this method differently